test/Microsoft.AspNet.Mvc.Core.Test/Routing/InnerAttributeRouteTest.cs

```diff
@@ -361,6 +361,63 @@ public class InnerAttributeRouteTest
         }
 
         [Theory]
+        [InlineData("template", "template/{parameter}", "/template/5", true)]
+        [InlineData("template", "template/{parameter}", "/template", false)]
+        [InlineData("template/{parameter}", "template/{parameter}/{id}", "/template/5/1234", true)]
+        [InlineData("template", "template/{parameter:int=1033}", "/template", false)]
+        [InlineData("template", "template/{parameter:int=5}", "/template/5", true)]
+        public void AttributeRoute_GenerateLink_UseAvailableVariables(
+            string firstTemplate,
+            string secondTemplate,
+            string expectedPath,
+            bool useValues)
+        {
+            // Arrange
+            var expectedGroup = CreateRouteGroup(0, useValues ? secondTemplate : firstTemplate);
```

Comment:
Yeah, I definitely feel a little gross doing conditionals like that, but have usually erred on the side of not duplicating code just because of it in the past, but I'll split it out if that's the general opinion.